pupil_src/shared_modules/gaze_producer/model/calibration_storage.py

```diff
@@ -114,7 +118,14 @@ def add(self, calibration, overwrite=False):
                 + "\n".join(f"- {c.name} ({c.unique_id})" for c in self._calibrations)
             )
             return
-
+        registered_gazer_class_names = set(
+            gazer_classes_by_class_name(user_selectable_gazer_classes()).keys()
+        )
+        if calibration.gazer_class_name not in registered_gazer_class_names:
+            logger.warning(
+                f"Did not add calibration {calibration.name} ({calibration.unique_id}) because gaze mapping method ({calibration.gazer_class_name}) is not available."
+            )
+            return
```

Comment:
Seeing this implementation, I do not see a reason for disallowing non-editable recordings with `GazerHMD3D` gazers. First, users can use their recorded hmd calibrations to apply a manual offset. At the same time, they would not be able to load or create an editable calibration that is set to "HMD 3D".
```suggestion

        is_calib_editable = (
            self._from_same_recording(calibration)
            and calibration.is_offline_calibration
        )
        if is_calib_editable:
            available_gazer_classes = user_selectable_gazer_classes()
        else:
            available_gazer_classes = registered_gazer_classes()

        gazer_class_names = gazer_classes_by_class_name(available_gazer_classes).keys()
        if calibration.gazer_class_name not in gazer_class_names:
            logger.warning(
                f"Did not add calibration {calibration.name} ({calibration.unique_id}) "
                f"because gaze mapping method ({calibration.gazer_class_name}) is not "
                f"available."
            )```
This change requires importing `registered_gazer_classes` from `gaze_mapping`

Reply:
Makes sense to me.